src/lib/concurrency/transaction_context.hpp

```diff
@@ -81,14 +90,17 @@ class TransactionContext : public std::enable_shared_from_this<TransactionContex
   TransactionPhase phase() const;
 
   /**
-   * Returns true if transaction has been aborted.
+   * Returns true if transaction has been aborted due to an error.
```

Comment:
Same as before: We don't really have errors. If this is only for RollBackAfterConflict, maybe rename the method? Above you have used "failed", maybe that helps?